src/Container/DynamicInitializer.php

```diff
@@ -4,7 +4,9 @@
 
 namespace Tempest\Container;
 
-interface CanInitialize
+interface DynamicInitializer
 {
     public function canInitialize(string $className): bool;
+
+    public function initialize(string $className, Container $container): object;
 }
```

Comment:
To me, something like an `#[InitializesAny(Request::class)]` attribute makes sense to me here, but I may just be unduly biased toward attributes. 😛

Reply:
No that's actually a good alternative. I wanna consider that after merging this PR.

I tried to get rid of the interface and do `is_a` mapping on all dependencies, but that quickly lead to many recusive edge cases…

Reply:
https://github.com/tempestphp/tempest-framework/issues/188